packages/data-provider/src/utils/auditUtils.ts

```diff
@@ -4,19 +4,9 @@ import { z } from 'zod';
 import { AUDIT_ACTION, AuditAction, AuditDataResponse, AuditRepositoryRecord } from './types.js';
 
 /**
- * Convert Audit Action enum into an array of strings in uppercase to facilitate validation
+ * Create Zod enum schema transforming to uppercase enum values
  */
-const upperCaseAuditActions: string[] = Object.values(AUDIT_ACTION).map((value) => value.toUpperCase());
-
-/**
- * This function takes an array of values and casts it to a tuple type where the first element is of the same type as the array, and the rest are of the same type as well.
- */
-const zodEnum = <T>(arr: T[]): [T, ...T[]] => arr as [T, ...T[]];
-
-/**
- * Create Zod enum schema from uppercase enum values
- */
-const auditActionSchema = z.enum(zodEnum(upperCaseAuditActions));
+const auditActionSchema = z.nativeEnum(AUDIT_ACTION).transform((value) => value.toUpperCase());
```

Comment:
You can remove the toUpperCase here to make sure that `auditActionSchema` parses into the enum string literals. You perform the `toUpperCase` conversion in `isAuditEventValid` and `convertToAuditEvent` before parsing anyways. My objective here is to be able to remove the `as AUDIT_ACTION` on the `convertToAuditEvent` return statement.

Reply:
converted original `AUDIT_ACTION` to zod enum to simplify validations